src/raster/r.suitability.regions/r.suitability.regions.py

```diff
@@ -0,0 +1,410 @@
+#
+#            This program is free software under the GNU General Public
+#            License (>=v2). Read the file COPYING that comes with GRASS
+#            for details.
+#
+##############################################################################
+
+#%module
+#% description: From suitability map to suitable regions
+#%end
+#%option G_OPT_R_INPUT
+#% key: suitrast
+#% label: Suitability raster
+#% description: Raster layer represting suitability (0-1)
+#% required: yes
+#% multiple: no
+#% guisection: Input
+#%end
+#%option G_OPT_R_OUTPUT
+#% key: suitreg
+#% label: Output raster
+#% description: Raster with candidate regions for conservation
+#% required: yes
+#% multiple: no
+#% guisection: Input
+#%end
+#%option
+#% key: minsuit
+#% description: Threshold value above which areas are considered suitable
+#% required: yes
+#% type: string
+#% answer: 0.7
+#% key_desc: float
+#% guisection: Input
+#%end
+#%option
+#% key: minfrag
+#% description: Minimum area (in hectares) limit for fragments
+#% required: yes
+#% type: string
+#% answer: 1000
+#% key_desc: float
+#% guisection: Input
+#%end
+#%option
+#% key: absminsuit
+#% description: Threshold below which cells are marked as fully unsuitable
+#% required: no
+#% type: string
+#% key_desc: float
+#% guisection: Input
+#%end
+
+#%flag
+#% key: d
+#% label: Clumps including diagonal neighbors
+#% description: Clumps including diagonal neighbors
+#% guisection: Optional
+#%end
+
+#%option
+#% key: radius
+#% description: Radius for focal stat
+#% required: no
+#% type: integer
+#% answer: 1
+#% guisection: Focal stats
+#%end
+#%flag
+#% key: c
+#% label: Circular neighborhood
+#% description: Use circular neighborhood for focal statistic
+#% guisection: Focal stats
+#%end
+#%option
+#% key: focalstat
+#% description: Neighborhood operation (focal statistic)
+#% required: no
+#% type: string
+#% answer: median
+#% options:maximum,median,quart1,quart3
+#% guisection: Focal stats
+#%end
+
+#%option
+#% key: maxgap
+#% description: Maximum gap size (in hectares) to remove
+#% required: no
+#% type: string
+#% answer: 0
+#% key_desc: float
+#% guisection: Remove gaps
+#%end
+
+#%flag
+#% key: z
+#% label: Suitability per region
+#% description: Compute map with average suitability per region
+#% guisection: Reporting stats
+#%end
+#%flag
+#% key: a
+#% label: Calculates area of clumped areas (hectares)
+#% description: Calculates area of clumped areas
+#% guisection: Reporting stats
+#%end
+
+#%flag
+#% key: k
+#% label: Keep map with all suitable cells
+#% description: Keep map that show suitable areas (irrespective of clump size)
+#% guisection: Optional
+#%end
+
+#%flag
+#% key: f
+#% label: Keep suitability based on focal statistics
+#% description: Keep suitabiliyt based on focal statistics
+#% guisection: Optional
+#%end
+
+import sys
+import os
+import atexit
+import uuid
+import grass.script as gs
+
+RECLASSRULES1 = """\
+1:filled gaps
+2:suitable areas
+"""
+COLORRULES1 = """
```

Comment:
Purpose or another description in name would be better for readability instead of numbering those. Underscore to separate words.